build.sh

```diff
@@ -0,0 +1,55 @@
+######################################################################
+# Copyright (c) 2019 Claudio André <claudioandre.br at gmail.com>
+#
+# This program comes with ABSOLUTELY NO WARRANTY; express or implied.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, as expressed in version 2, seen at
+# http://www.gnu.org/licenses/gpl-2.0.html
+######################################################################
+
+#!/bin/bash
+
+# Build options (system wide, disable checks, etc.)
+SYSTEM_WIDE='--with-systemwide --enable-rexgen'
+X86_REGULAR="--disable-native-tests --disable-opencl $SYSTEM_WIDE"
+X86_NO_OPENMP="--disable-native-tests --disable-opencl $SYSTEM_WIDE --disable-openmp"
+
+OTHER_REGULAR="$SYSTEM_WIDE"
+OTHER_NO_OPENMP="$SYSTEM_WIDE --disable-openmp"
+
+# Build helper
+function do_build () {
+    set -e
+
+    if [[ -n "$1" ]]; then
+        make -s clean && make -sj4 && mv ../run/john "$1"
+    else
+        make -s clean && make -sj4
+    fi
+    set +e
+}
+
+#if (Build); then
+    echo ""
+    echo "---------------------------- BUILDING -----------------------------"
+
+    if [[ "$(uname -m)" == "x86_64" || "$(uname -m)" == "i386" || "$(uname -m)" == "i686" ]]; then
```

Comment:
You can achieve that with config-opts.

Reply:
If you don't mind, I would like to keep this as is (using a shell script):
* I have to set the build options (ok). But;
  * I have to rename the binary;
  * I have to inform the next binary the name of the previous one, escaping whatever needed.
  * Some binaries fallback to OpenMP, some to an extension plus OpenMP;

Plus, that I'm sharing the script with other build systems: snap and Windows. Well, and I saw no `config-opts` use-case complex as this one.

Reply:
Looks like that's what more or less other distributions are doing so I guess we will have to live with it.